### PR TITLE
create revision

### PR DESCRIPTION
**Pull request by Araballi**
create revision

### CODE REVIEW

File: src/Kmd.Logic.Gateway.Automation/Publish.cs [hunk 1]
```diff
@@ -232,6 +232,36 @@ private async Task CreateApi(IGatewayClient client, Guid subscriptionId, Guid pr
...
+                await this.CreateRevision(client, subscriptionId, apiVersionId, folderPath, revision).ConfigureAwait(false);
+            }
+        }
+
+        private async Task CreateRevision(IGatewayClient client, Guid subscriptionId, Guid apiVersionId, string folderPath, Revision revision)
+        {
+            using var revisionOpenApiSpec = new FileStream(path: Path.Combine(folderPath, revision.OpenApiSpecFile), FileMode.Open);
+            var revisionResponse = await client.CreateRevisionAsync(
+                subscriptionId: subscriptionId,
+                apiId: apiVersionId,
```
Comment (jaceks2106) [inline src/Kmd.Logic.Gateway.Automation/Publish.cs:257]: Are you sure that `apiVersionId` should be assigned to `apiId`?

**Review by jaceks2106 — CHANGES_REQUESTED** [review]

File: src/Kmd.Logic.Gateway.Automation/Publish.cs [hunk 2]
```diff
@@ -232,6 +232,36 @@ private async Task CreateApi(IGatewayClient client, Guid subscriptionId, Guid pr
...
+        }
+
+        private async Task CreateRevisions(IGatewayClient client, Guid subscriptionId, Guid apiVersionId, string folderPath, IEnumerable<Revision> revisions)
+        {
+            if (revisions == null)
+            {
+                return;
+            }
+
+            foreach (var revision in revisions)
```
Comment (jaceks2106) [inline src/Kmd.Logic.Gateway.Automation/Publish.cs:246]: We can run this in parallel:

```cs
return Task.WhenAll(revisions.Select(r =>this.CreateRevision(client, subscriptionId, apiVersionId, folderPath, r)));
```

Comment (jaceks2106) [reply src/Kmd.Logic.Gateway.Automation/Publish.cs:246]: And then:

```cs
if (revisions == null)
            {
                return Task.CompletedTask;
            }
```

File: src/Kmd.Logic.Gateway.Automation/Publish.cs [hunk 3]
```diff
@@ -232,6 +232,36 @@ private async Task CreateApi(IGatewayClient client, Guid subscriptionId, Guid pr
...
+            {
+                await this.CreateRevision(client, subscriptionId, apiVersionId, folderPath, revision).ConfigureAwait(false);
+            }
+        }
+
+        private async Task CreateRevision(IGatewayClient client, Guid subscriptionId, Guid apiVersionId, string folderPath, Revision revision)
+        {
+            using var revisionOpenApiSpec = new FileStream(path: Path.Combine(folderPath, revision.OpenApiSpecFile), FileMode.Open);
```
Comment (jaceks2106) [inline src/Kmd.Logic.Gateway.Automation/Publish.cs:254]: Add at the end `, FileAccess.Read`

File: src/Kmd.Logic.Gateway.Automation/Publish.cs (hunk 1 above)
Comment (Araballi) [reply src/Kmd.Logic.Gateway.Automation/Publish.cs:257]: yes apiversionId used for revision

Comment (jaceks2106) [reply src/Kmd.Logic.Gateway.Automation/Publish.cs:257]: But it shouldn't be called apiVersionId, but rather apiId.

**Review by jaceks2106 — APPROVED** [review]

**Pull request merged**